modules/exoplanets/tasks/exoplanetscsvtobintask.cpp

```diff
@@ -0,0 +1,677 @@
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#include <modules/exoplanets/tasks/exoplanetscsvtobintask.h>
+
+#include <modules/exoplanets/exoplanetshelper.h>
+#include <openspace/documentation/documentation.h>
+#include <openspace/documentation/verifier.h>
+#include <ghoul/filesystem/filesystem.h>
+#include <ghoul/fmt.h>
+#include <ghoul/glm.h>
+#include <ghoul/logging/logmanager.h>
+#include <ghoul/misc/dictionary.h>
+#include <fstream>
+
+namespace {
+    constexpr const char* KeyInputCsv = "InputCSV";
+    constexpr const char* KeyInputSpeck = "InputSPECK";
+    constexpr const char* KeyOutputBin = "OutputBIN";
+    constexpr const char* KeyOutputLut = "OutputLUT";
+
+    constexpr const char* _loggerCat = "CsvToBinTask";
+
+    constexpr const char* TeffBvPath = "${SYNC}/http/exoplanets_data/1/teff_bv.txt";
+} // namespace
+
+namespace openspace::exoplanets {
+
+ExoplanetsCsvToBinTask::ExoplanetsCsvToBinTask(const ghoul::Dictionary& dictionary) {
+    openspace::documentation::testSpecificationAndThrow(
+        documentation(),
+        dictionary,
+        "ExoplanetsCsvToBinTask"
+    );
+
+    _inputCsvPath = absPath(dictionary.value<std::string>(KeyInputCsv));
+    _inputSpeckPath = absPath(dictionary.value<std::string>(KeyInputSpeck));
+    _outputBinPath = absPath(dictionary.value<std::string>(KeyOutputBin));
+    _outputLutPath = absPath(dictionary.value<std::string>(KeyOutputLut));
+}
+
+std::string ExoplanetsCsvToBinTask::description() {
+    return "Extract metadata from csv-file " + _inputCsvPath +
+        " and write as bin to " + _outputBinPath;
+}
+
+void ExoplanetsCsvToBinTask::perform(const Task::ProgressCallback& progressCallback) {
+    std::ifstream csvFile(_inputCsvPath);
+    if (!csvFile.good()) {
+        LERROR(fmt::format("Failed to open CSV file '{}'", _inputCsvPath));
+        return;
+    }
+
+    std::ofstream binFile(_outputBinPath, std::ios::out | std::ios::binary);
+    std::ofstream lutFile(_outputLutPath);
+
+    int version = 1;
+    binFile.write(reinterpret_cast<char*>(&version), sizeof(int));
+
+    Exoplanet p;
+
+    std::string planetName;
+    std::string component;
+    std::string planetRow;
+    getline(csvFile, planetRow); // The first line, containing the data names
+
+    bool isKeplerObject = false;
+    int total = 0;
+    while (getline(csvFile, planetRow)) {
+        ++total;
+    }
+    csvFile.clear();
+    csvFile.seekg(0);
+    getline(csvFile, planetRow); // The first line, containing the data names
+    LINFOC("CSVTOBIN", fmt::format("Loading {} stars", total));
+
+    auto readFloatData = [](const std::string& data) -> float {
```

Comment:
If you want to be a bit C++-fancier, you can use `from_chars` from the `<utility> header:
```
int result;
auto [p, ec] = std::from_chars(data.begin(), data.end(), result);
if (ec == std::errc()) {
  return result;
}
else {
   return NAN;
}
```

Reply:
https://en.cppreference.com/w/cpp/language/structured_binding